ruby/startupinabox/app/services/dependencies/uploader.rb

```
@@ -0,0 +1,35 @@
+require 'securerandom'
+
+module Services
+  module Dependencies
+    class Uploader
+
+      def initialize(client, bucket_name)
+        @client = client
+        @bucket_name = bucket_name
+        @bucket = Aws::S3::Bucket.new(bucket_name, client: @client)
+      end
+
+      def upload(file)
+        key = SecureRandom.uuid
+        obj = @bucket.object(key)
+        return obj.upload_file(file) && key
```

Maybe there is a version of this method in the API that raises an exception instead of just returning `false`?